<?xml version="1.0"?>
<launch>

    <arg name="id" default="1" />
    <arg name="frame" />
    <arg name="x" default="0" />
    <arg name="y" default="0" />
    <arg name="output" default="log" />

    <!-- Position controller -->
    <node pkg="navigation_pos_controller" type="navigation_pos_controller" name="navigation_pos_controller_$(arg id)" output="$(arg output)" required="true">
        <param name="x" value="$(arg x)" />
        <param name="y" value="$(arg y)" />
        <rosparam file="$(find navigation_pos_controller)/param/navigation_pos_controller.yaml" command="load" />
    </node>

    <!-- Navigation move base -->
    <node pkg="move_base" type="move_base" name="move_base_$(arg id)" output="$(arg output)" required="true">
        <rosparam file="$(find navigation_pos_controller)/param/common_costmap.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find navigation_pos_controller)/param/common_costmap.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find navigation_pos_controller)/param/navigation.yaml" command="load" />
        <rosparam file="$(find navigation_pos_controller)/param/local_costmap.yaml" command="load" />
        <rosparam file="$(find navigation_pos_controller)/param/global_costmap.yaml" command="load" />
        <param name="local_costmap/robot_base_frame" value="$(arg frame)/base_link" />
        <param name="local_costmap/global_frame" value="$(arg frame)/odom" />
        <param name="global_costmap/robot_base_frame" value="$(arg frame)/base_link" />
        <param name="global_costmap/global_frame" value="map" />
<!--         <remap from="map" to="/map" /> -->
    </node>

</launch>
